### PR TITLE
Add an endpoint to get all donors with flairs

### PR DESCRIPTION
**Pull request by amCap1712**

### CODE REVIEW

File: listenbrainz/webserver/views/donor_api.py [hunk 1]
```diff
@@ -38,3 +39,19 @@ def biggest_donors():
 
     donors = get_biggest_donors(meb_conn, db_conn, count, offset)
     return jsonify(donors)
+
+
+@donor_api_bp.route("/all-flairs", methods=["GET", "OPTIONS"])
+@crossdomain
+@ratelimit()
+def all_flairs():
+    """ Get flairs (including expired ones) for all users """
```
Comment (mayhem) [inline listenbrainz/webserver/views/donor_api.py:48]: I think we need more details in this endpoint documentation. What does this endpoint return? Does the caller need to care about expired flairs?

File: listenbrainz/webserver/views/user_settings_api.py [hunk 2]
```diff
@@ -27,6 +27,8 @@ def update_flair():
     if "flair" not in request.json:
         raise APIBadRequest("Missing flair")
 
+    # todo: add flair validation
```
Comment (mayhem) [inline listenbrainz/webserver/views/user_settings_api.py:30]: Does this need to be done now or add a ticket?

**Review by mayhem — CHANGES_REQUESTED** [review]

File: listenbrainz/webserver/views/user_settings_api.py (hunk 2 above)
Comment (amCap1712) [reply listenbrainz/webserver/views/user_settings_api.py:30]: this needs to be done (if at all) after the structure of the flairs has been decided.

File: listenbrainz/webserver/views/donor_api.py (hunk 1 above)
Comment (amCap1712) [reply listenbrainz/webserver/views/donor_api.py]: fixed.

File: listenbrainz/webserver/views/user_settings_api.py (hunk 2 above)
Comment (mayhem) [reply listenbrainz/webserver/views/user_settings_api.py:30]: OK, sounds like no validation will be needed, so lets remove the comment on move on.

**Review by mayhem — APPROVED** [review]

**Pull request merged**